In [25]:
import pandas as pd
import numpy as np
from sklearn import linear_model

In [26]:
df = pd.read_csv('home_prices.csv')
df

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       NaN   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000

In [27]:
df.bedrooms.median()

3.5

In [28]:
# Data Preprocessing

import math
median_bedrooms = math.floor(df.bedrooms.median())
median_bedrooms

3

In [29]:
df.bedrooms.fillna(median_bedrooms)

0    3.0
1    4.0
2    3.0
3    3.0
4    5.0
Name: bedrooms, dtype: float64

In [30]:
df.bedrooms = df.bedrooms.fillna(median_bedrooms)
df

area  bedrooms  age   price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       3.0   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000

In [31]:
# Trainig of the Model

reg = linear_model.LinearRegression()
reg.fit(df[['area', 'bedrooms', 'age']], df.price)

LinearRegression()

In [32]:
# Coefficient of the Data

reg.coef_

array([   137.25, -26025.  ,  -6825.  ])

In [33]:
# Intercept of the Data

reg.intercept_

383725.0

**Linear Regression**

**price = (137.25 * area) + (-26025 * bedrooms) + (-6825 * age) + 383725.0**

 Predecting a House with the Dimension **3000 Sq.Ft, 3 Bedrooms, 40 Year Old**

In [39]:
reg.predict([[3000, 3, 40]])

array([444400.])

Prediceted price of the House is **444400.**

Predecting a House with the Dimension **2500 Sq.Ft, 4 Bedrooms, 5 Year Old**

In [40]:
reg.predict([[2500, 4, 5]])

array([588625.])

Prediceted price of the House is **588625.**